setup.py

```diff
@@ -20,35 +19,26 @@ def read(f):
 
 
 def read_version():
-    regexp = re.compile(r"^__version__\W*=\W*'([\d.abrc]+)'")
-    init_py = os.path.join(os.path.dirname(__file__),
-                           'aiomonitor', '__init__.py')
-    with open(init_py) as f:
-        for line in f:
-            match = regexp.match(line)
-            if match is not None:
-                return match.group(1)
-        else:
-            msg = 'Cannot find version in aiomonitor/__init__.py'
-            raise RuntimeError(msg)
+    return version('aiomonitor')
 
 
 classifiers = [
     'License :: OSI Approved :: MIT License',
     'Intended Audience :: Developers',
     'Programming Language :: Python :: 3',
-    'Programming Language :: Python :: 3.5',
-    'Programming Language :: Python :: 3.6',
-    'Programming Language :: Python :: 3.7',
     'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
     'Operating System :: POSIX',
     'Development Status :: 3 - Alpha',
     'Framework :: AsyncIO',
 ]
 
 
 setup(name='aiomonitor',
-      version=read_version(),
+      use_scm_version=True,
```

Comment:
Modern `setuptools-scm` relies on the `pyproject.toml`-based config. So we'll need to add respective settings there. `setup.py` should be phased out, at least, partially.
Though, I'd start by performing the conversion gradually.
First, convert most static bits to `setup.cfg` — the `setup.py` module remains but without the moved settings/metadata. It's possible to source version and readme with this config using `attr:` and `file:` helpers without any Python code at all.
Then, enable PEP 517 via `pyproject.toml` which would let you set minimum required setuptools, among other things.
Starting with a certain setuptools version, you can even delete the entire `setup.py`, only having `setup.cfg`+`pyproject.toml`.
Then, you can switch to setuptools-scm for the dist meta and `importlib.metadata` to read that in runtime. Alternatively, setuptools-scm can also generate a gitignored Python module on build with version informacion importable as normal.

One more thing — to make the Git archives (through GitHub URLs mostly) installable, make sure to set up the `.git_archival.txt` file in the repo per the official instructions.

Reply:
Added `.git_archival.txt`, I'll leave this open so we can return to it later.